### PR TITLE
Reformat and promisification of hyperdb functions

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

**Comment by G-Ray** [comment]
Yes, definitely we should convert the hyperdb callbacks to promises to get a cleaner.
As discussed, can you move these methods to a standalone file like `utils.js` ?

File: src/library/masq.js [hunk 1]
```diff
@@ -221,69 +198,49 @@ class Masq {
...
+    )
+    const resourceNodes = await Promise.all(resourcePromises)
+    const resources = resourceNodes.map(n => n.value)
+    return resources
   }
 
-  _updateResource (profileId, name, res) {
-    if (!profileId) return Error('missing profileId')
+  async _updateResource (profileId, name, res) {
+    if (!profileId) throw Error('missing profileId')
```
Comment (G-Ray) [inline src/library/masq.js:238]: :+1:

File: src/library/masq.js [hunk 2]
```diff
@@ -221,69 +198,49 @@ class Masq {
...
-      })
-    })
+    const node = await hyperdbUtils.promiseGet(this.dbs.core, `/profiles/${profileId}/${name}`)
+    if (!node) return []
+
+    const ids = node.value
+    const resourcePromises = ids.map(
+      id => hyperdbUtils.promiseGet(this.dbs.core, `/profiles/${profileId}/${name}/${id}`)
+    )
+    const resourceNodes = await Promise.all(resourcePromises)
```
Comment (G-Ray) [inline src/library/masq.js:232]: :+1:

**Review by G-Ray — APPROVED** [review]

**Comment by G-Ray** [comment]
Nice, code is cleaner with these promises, approved

**Review by leventdem — APPROVED** [review]
Nice. Add getApp in the future.

**Pull request merged**